dfu_target SUIT cache implementation and unit tests. (#17281)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

DFU libraries
-------------

* :ref:`lib_dfu_target` library:

>   Updated the DFU Target SUIT implementation to the newest version of the SUIT.
>   Added SUIT Cache processing to the DFU Target SUIT library.
>   You can learn more about the DFU Target SUIT implementation in the :ref:`lib_dfu_target_suit_style_update` section.

Reviewer: ```suggestion
  - Updated the DFU Target SUIT implementation to the newest version of the SUIT.
  - Added SUIT Cache processing to the DFU Target SUIT library, as described in the :ref:`lib_dfu_target_suit_style_update` section.
```
